transport/client.go

```diff
@@ -0,0 +1,135 @@
+package transport
+
+import (
+	"buf.build/gen/go/meshtastic/protobufs/protocolbuffers/go/meshtastic"
+	"fmt"
+	"github.com/charmbracelet/log"
+	"google.golang.org/protobuf/proto"
+	"math/rand"
+)
+
+type HandlerFunc func(message proto.Message)
+
+type Client struct {
+	sc       *StreamConn
+	handlers *HandlerRegistry
+	log      *log.Logger
+
+	config struct {
+		complete bool
+		configID uint32
+		*meshtastic.MyNodeInfo
+		*meshtastic.DeviceMetadata
+		nodes    []*meshtastic.NodeInfo
+		channels []*meshtastic.Channel
+		config   []*meshtastic.Config
+		modules  []*meshtastic.ModuleConfig
+	}
+}
+
+func NewClient(sc *StreamConn, errorOnNoHandler bool) *Client {
+	return &Client{
+		log:      log.WithPrefix("client"),
+		sc:       sc,
+		handlers: NewHandlerRegistry(errorOnNoHandler),
+	}
+}
+
+// You have to send this first to get the radio into protobuf mode and have it accept and send packets via serial
+func (c *Client) sendGetConfig() error {
+	r := rand.Uint32()
+	c.config.configID = r
+	msg := &meshtastic.ToRadio{
+		PayloadVariant: &meshtastic.ToRadio_WantConfigId{
+			WantConfigId: r,
+		},
+	}
+	c.log.Debug("sending want config", "id", r)
+	if err := c.sc.Write(msg); err != nil {
+		return fmt.Errorf("writing want config command: %w", err)
+	}
+	c.log.Debug("sent want config")
+	return nil
+}
+
+func (c *Client) Handle(kind proto.Message, handler MessageHandler) {
+	c.handlers.RegisterHandler(kind, handler)
+}
+
+func (c *Client) SendToRadio(msg *meshtastic.ToRadio) error {
+	return c.sc.Write(msg)
+}
+
+func (c *Client) Connect() error {
```

Comment:
I'm not super happy with the Connect() method spinning off a goroutine we can't yet close - but this is the existing implementation and I didn't want to change it too much.

Reply:
Agreed, it should close gracefully. TODO 😆